scalding-core/src/main/scala/com/twitter/scalding/reducer_estimation/Common.scala

```diff
@@ -123,26 +177,52 @@ object ReducerEstimatorStepStrategy extends FlowStepStrategy[JobConf] {
 /**
  * Info about a prior FlowStep, provided by implementers of HistoryService
  */
-sealed trait FlowStepHistory {
-  /** Size of input to mappers (in bytes) */
-  def mapperBytes: Long
-  /** Size of input to reducers (in bytes) */
-  def reducerBytes: Long
-}
-
-object FlowStepHistory {
-  def apply(m: Long, r: Long) = new FlowStepHistory {
-    override def mapperBytes: Long = m
-    override def reducerBytes: Long = r
-  }
-}
+sealed case class FlowStepHistory(keys: FlowStepKeys,
```

Comment:
Whoops, I didn't realize that Scala has a final keyword!
I think final would be more appropriate.